src/commands/protocols/update/receiver/submit-update-commit-command.js

```diff
@@ -0,0 +1,127 @@
+import EpochCommand from '../../common/epoch-command.js';
+import {
+    OPERATION_ID_STATUS,
+    ERROR_TYPE,
+    COMMAND_RETRIES,
+    BLOCK_TIME,
+} from '../../../../constants/constants.js';
+
+class SubmitUpdateCommitCommand extends EpochCommand {
+    constructor(ctx) {
+        super(ctx);
+        this.blockchainModuleManager = ctx.blockchainModuleManager;
+        this.serviceAgreementService = ctx.serviceAgreementService;
+        this.operationIdService = ctx.operationIdService;
+        this.shardingTableService = ctx.shardingTableService;
+        this.networkModuleManager = ctx.networkModuleManager;
+
+        this.errorType = ERROR_TYPE.COMMIT_PROOF.SUBMIT_UPDATE_COMMIT_ERROR;
+    }
+
+    async execute(command) {
+        const {
+            blockchain,
+            contract,
+            tokenId,
+            keyword,
+            hashFunctionId,
+            agreementData,
+            agreementId,
+            operationId,
+        } = command.data;
+
+        const epoch = await this.calculateCurrentEpoch(
+            agreementData.startTime,
+            agreementData.epochLength,
+            blockchain,
+        );
+
+        const stateIndex = await this.blockchainModuleManager.getAssertionIdsLength(
+            blockchain,
+            contract,
+            tokenId,
+        );
+        if (command.retries === COMMAND_RETRIES.SUBMIT_UPDATE_COMMIT) {
+            this.operationIdService.emitChangeEvent(
+                OPERATION_ID_STATUS.COMMIT_PROOF.SUBMIT_UPDATE_COMMIT_START,
+                operationId,
+                agreementId,
+                epoch,
+            );
+        }
+
+        this.logger.trace(
+            `Started ${command.name} for agreement id: ${command.data.agreementId} ` +
+                `blockchain: ${blockchain} contract: ${contract}, token id: ${tokenId}, ` +
+                `keyword: ${keyword}, hash function id: ${hashFunctionId}, stateIndex: ${stateIndex}. Retry number ${
+                    COMMAND_RETRIES.SUBMIT_UPDATE_COMMIT - command.retries + 1
+                }`,
+        );
+
+        const hasPendingUpdates = await this.blockchainModuleManager.isUpdateCommitWindowOpen(
```

Comment:
Wrong check for pending update

Reply:
why? it's that same as it is in contract